spec/tic_tac_toe_spec.rb

```diff
@@ -0,0 +1,206 @@
+def find_winner(board)
+    # returns row winner
+    board.each do |row|
+        row_winner = find_winner_in_line(row)
+        if row_winner
+            return row_winner
+        end
+    end
+
+    # returns column winner
+    board.transpose.each do |col|
+        col_winner = find_winner_in_line(col)
+        if col_winner
+            return col_winner
+        end
+    end
+
+    # returns diagonal winner
+    if find_winner_in_diagonal(board)
+        return find_winner_in_diagonal(board)
+    end
+
+    # check for any unfilled space
+    board.each do |row|
+        row.each do |col|
+            if col == '-'
+                return false
+            end
+        end
+    end
+
+    :tie
+end
+
+def find_winner_in_line(row)
+
+    if row.count('X') == 3
+        'X'
+    elsif row.count('O') == 3
+        'O'
+    else
+        false
+    end
+
+end
+
+def find_winner_in_diagonal(board)
+    diagonal1 = [board[0][0], board[1][1], board[2][2]]
+    diagonal2 = [board[0][2], board[1][1], board[2][0]]
+
+    if find_winner_in_line(diagonal1)
+        return find_winner_in_line(diagonal1)
```

Comment:
Here's another case where you can save the computed value from `find_winner_in_line` in a local variable instead of calling it twice with the same parameters.